File: Cenfomon/Capa_Logica/src/main/java/com/cenfomon/Cenfomon_Capa_Logica/Expansion.java
package com.cenfomon.Cenfomon_Capa_Logica;

public class Expansion extends Items {

    private int beltUpgrade;
    private int idUsuario;


    public Expansion(int id, String location, float cost, String name, String imgURL, int beltUpgrade, int idUsuario) {
        super(id, location, cost, name, imgURL);
        this.beltUpgrade = beltUpgrade;
        this.idUsuario = idUsuario;
    }


    public int getBeltUpgrade() {
        return beltUpgrade;
    }

    public void setBeltUpgrade(int beltUpgrade) {
        this.beltUpgrade = beltUpgrade;
    }

    public int getIdUsuario() {
        return idUsuario;
    }

    public void setIdUsuario(int idUsuario) {
        this.idUsuario = idUsuario;
    }
}

File: Cenfomon/Cenfomon_COI/bl/factoryMethodItems/productoConcreto/Expansion.ts
import {Item} from "../Producto/Item";

export class Expansion extends Item{


    constructor(_id: number, _location: Location, _cost: number,_name: string, _imgURL: string, private _beltUpgrade: number, private _idUsuario: number) {
        super(_id, _location , 60 ,_name ,_imgURL);
        this._beltUpgrade = _beltUpgrade;
        this._idUsuario = _idUsuario;
    }

    generarID(): {} {
        return {2000:+((Math.random() * 999))};
    }


    get beltUpgrade(): number {
        return this._beltUpgrade;
    }

    set beltUpgrade(value: number) {
        this._beltUpgrade = value;
    }

    get idUsuario(): number {
        return this._idUsuario;
    }

    set idUsuario(value: number) {
        this._idUsuario = value;
    }
}